From 08fdaedc9f006198a350f0dae76fec8346f2e1ef Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:32 -0500
Subject: [drm] i915: don't try to retrain a DP link on an inactive CRTC

Message-id: <1418852380-13061-22-git-send-email-rclark@redhat.com>
Patchwork-id: 102157
O-Subject: [RHEL7 drm 21/89] drm/i915: don't try to retrain a DP link on an inactive CRTC
Bugzilla: 1153301
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1153301
Upstream: since v3.16.4

commit a81dc68e37edb97cfd6eddbc4b5a594a123519ac

Author:     Imre Deak <imre.deak@intel.com>
AuthorDate: Mon Aug 18 14:42:46 2014 +0300
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:13:12 2014 -0500

    drm/i915: don't try to retrain a DP link on an inactive CRTC

    commit 1a125d8a2c22b11741fc47d4ffcf7a5ffa044dd3 upstream.

    Atm we may retrain the DP link even if the CRTC is inactive through
    HPD work->intel_dp_check_link_status(). This in turn can lock up the PHY
    (at least on BYT), since the DP port is disabled.

    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=81948
    Signed-off-by: Imre Deak <imre.deak@intel.com>
    Reviewed-by: Ville Syrjala <ville.syrjala@linux.intel.com>
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 5d084f0..7ad5f0e 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -3413,6 +3413,9 @@ intel_dp_check_link_status(struct intel_dp *intel_dp)
  if (WARN_ON(!intel_encoder->base.crtc))
   return;
 
+ if (!to_intel_crtc(intel_encoder->base.crtc)->active)
+  return;
+
  /* Try to read receiver status if the link appears to be up */
  if (!intel_dp_get_link_status(intel_dp, link_status)) {
   return;
-- 
1.7.1